lib/handlers/compile.ts

```diff
@@ -144,15 +144,28 @@ export class CompileHandler {
 
             if (contentType === 'application/json') {
                 bodyData = JSON.stringify(body);
-            }
-
-            if (contentType === 'application/x-www-form-urlencoded') {
+            } else if (contentType === 'application/x-www-form-urlencoded') {
                 bodyData = body;
+            } else {
+                Sentry.captureException(
+                    new Error(`Unexpected Content-Type received by /compiler/:compiler/compile: ${contentType}`),
+                );
+                proxyReq.write('Unexpected Content-Type');
             }
 
-            if (bodyData) {
-                proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
-                proxyReq.write(bodyData);
+            try {
+                if (bodyData) {
+                    proxyReq.setHeader('Content-Length', Buffer.byteLength(bodyData));
+                    proxyReq.write(bodyData);
+                }
+            } catch (e: any) {
+                Sentry.captureException(e);
```

Comment:
Maybe very overkill

Reply:
if it's waking me up in the night sending me messages cos the site's crashing, it's not overkill :D